sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/parser/DDLParserSuite.scala

```diff
@@ -2854,10 +2856,139 @@ class DDLParserSuite extends AnalysisTest {
       exception = parseException(
         "CREATE TABLE my_tab(a INT, b INT GENERATED ALWAYS AS a + 1) USING PARQUET"),
       errorClass = "PARSE_SYNTAX_ERROR",
-      parameters = Map("error" -> "'a'", "hint" -> ": missing '('")
+      parameters = Map("error" -> "'a'", "hint" -> "")
     )
   }
 
+  test("SPARK-48824: implement parser support for GENERATED ALWAYS AS IDENTITY columns in tables") {
+    for {
+      identityColumnDefStr <- Seq("BY DEFAULT", "ALWAYS")
+      identityColumnSpecStr <- Seq(
+        "START WITH 1 INCREMENT BY 1",
+        "INCREMENT BY 1 START WITH 1",
+        "START WITH 1",
+        "INCREMENT BY 1",
```

Comment:
I'd like to see tests with non-default parameter values for the step and start so that we can test that code path more clearly. Negative steps and starts would also be interesting.